### ISSUE TITLE
Nativefier/Electron and Flash

### ISSUE THREAD

**Issue by waltder** [jiahaog/nativefier#944]
How dependent is Nativefier on Chrome?  My employer has a web application built on Flash and we have to support this app beyond 2020.  Nativefier API documentation for flash says " the version of Chrome on your computer should be the same as the version used by the version of Electron for the Nativefied package".  Does this mean that each computer I deploy my app to needs to have same version of Chrome that Electron uses, or just the computer I use to build the app?  

In order to build my Flash web application with Nativefier, do I need Chrome?  
In order to run my "Nativefied" Flash web application, do I need Chrome?

Specifications
Version of Nativefier (run $ nativefier --version): v8.0.6
Version of Node.js (run $ node --version): v12.9.1
OS: Windows 10
Architecture 64bit

**Comment by ronjouch** [comment]
Hi @waltder , I'm the current maintainer of Nativefier.

Flash under Nativefier is untested and I haven't heard about it in a _long_ time. It used to be an important feature, people did use it and report the occasional bug, but as you can expect with the de-Flash-isation of the modern web, it's no longer the case. It doesn't necessarily mean it's broken, so if you have financial/job incentive to take time looking at it, help welcome:

1. Look at docs.
2. Go spelunking in issues & PRs, including closed ones.
3. It's untested and Nativefier did go through a major refactor recently, so if Flash under 8.x is broken, give 7.x (latest is 7.7.1) a try.
4. Fiddle with Electron version (flag `-e`)
5. In 8.x, use the `--verbose` flag for more debug info.
5. Report your findings here, and PR/code docs welcome 🙂.

Now, to answer your questions:

- _"In order to build my Flash web application with Nativefier, do I need Chrome?"_
  → Nope, only the deployed environment needs Chrome, and the build environment needs to know where Chrome will live in the deployed environment.
- _"In order to run my "Nativefied" Flash web application, do I need Chrome?"_
  → Currently yes (see [deployed app runtime code in main.ts](https://github.com/jiahaog/nativefier/blob/master/app/src/main.ts#L35-L40)): it simply passes to electron the path of the plugin. However, if this machinery still works, I don't see anything intrinsically preventing you from copying the flash binary files in the app folder and adjusting the path in the app options defining the path (it's simply a json file written at build time). That could be a feature too, if you want to implement it.
- _" Does this mean that each computer I deploy my app to needs to have same version of Chrome that Electron uses, or just the computer I use to build the app?"_
  → Same as deployed host, for binary compatibility between Electron and Chrome. But I'm not sure how critical this is, fiddle with it and see. Also, see my answer to your previous question, maybe the PPAPI binaries can be copied in the app, letting you uninstall/upgrade Chrome on the machine where the nativefied app is deployed

On that, closing as your questions were answered I think, but feel free to come back with more questions, I'll see them.

**Comment by waltder** [comment]
Thanks ronjouch for your response.  I got my flash app to run with Nativefier.  I needed to provide the "flash-path" because the "auto-find" didn't work because it is looking in wrong directory.  My flash player was here: C:\Users\{userName}\AppData\Local\Google\Chrome\User Data\PepperFlash\32.0.0.344\pepflashplayer.dll.  Thanks for the info about the nativefier.json file.  The plan is to package the flash player with the app, and knowing I can make the change in that json file is good to know.  I did use the "electron-version" option too because version 10 has a fix for the PDF viewer.  The app runs great as a desktop application, with the one exception of opening non PDF files.  A blank window just opens and the file seems to download in the background.  I did look at the file download options, but none of those seem satisfactory.  I will probably get the Nativefier code to see if I can make a code change.

I was able to  uninstall Google Chrome (but I still had a copy of the flash player dll), shut down my computer, log back in again, and build and run my Nativefier flash app without Chrome.  Does that surprise you?
